modules/nux/modules.php

```diff
@@ -245,6 +245,293 @@ public function process() {
+                       $form['nux_config_profile_name'],
+                       $form['nux_config_imap_port'],
+                       $form['nux_config_email'],
+                       $form['nux_config_password'],
+                       $form['nux_config_imap_tls'],
+                       $form['nux_imap_sieve_host'],
+                       $form['nux_enable_sieve']);
+    
+                     if(!isset($result)) {
+                       /**
+                       * Delete the SMPT server created before
+                       **/
+                       if(!isset($result) && $form['nux_config_is_sender']){
+                           $res = Hm_SMTP_List::del($this->smtp_server_id);
+                           if ($res) {
+                               Hm_SMTP_List::forget_credentials($this->smtp_server_id);
+                               $smtp_servers = Hm_SMTP_List::dump(false, true);
+                               $this->user_config->set('smtp_servers', $smtp_servers);
+                           }
+                       }
+                        return;
+                     }
+                 }
+    
+                 if($form['nux_config_is_sender'] && $form['nux_config_is_receiver'] && $form['nux_create_profile'] && isset($this->imap_server_id) && isset($this->smtp_server_id)) {
+                     $this->saveProfile(
+                        $form['nux_config_profile_name'],
+                        $form['nux_profile_signature'],
+                        $form['nux_profile_reply_to'],
+                        $form['nux_profile_is_default'],
+                        $form['nux_config_email'],
+                        $form['nux_config_imap_address']
+                        );
+                 }
+    
+                 Hm_Msgs::add("Server saved");
+                 $this->out('just_saved_credentials', $this->just_saved_credentials);
+            }
+        }
+    }
+
+    public function saveProfile($name, $signature, $replyTo, $isDefault, $email, $serverMail){
+        $profile = array(
+                         'name' => $name,
+                         'sig' => $signature,
+                         'smtp_id' => $this->imap_server_id,
+                         'replyto' => $replyTo,
+                         'default' => $isDefault,
+                         'address' => $email,
+                         'server' =>  $serverMail,
+                         'user' => $email,
+                         'type' => 'imap'
+                     );
+
+          $profiles = new Hm_Profiles($this);
+          $profiles->add($profile);
+          $this->session->record_unsaved('Profile added');
+
+          $profiles->save($this->user_config);
+          $user_data = $this->user_config->dump();
+          $this->session->set('user_data', $user_data);
+    }
+
+    public function connectToIMAP($address, $name, $port, $user, $pass, $tls, $imap_sieve_host, $enableSieve = false, $errno = null, $errstr = null) {
```

Comment:
This and the following methods do too many things at once and are named misleadingly. They don't simply connect to a server but also check credentials, add error messages to the queue, save server credentials, etc. It is better to separate each task in distinct methods and also make sure the handler class handles the logic - error messages, calling saving server details code, etc.